### DISCUSSION TITLE
Different behavior between 2.x client and 4.x client when authenticate fails

### DISCUSSION THREAD

**Discussion by qiulang** [socketio/socket.io#4549] [General]
I notice a different behavior between 2.x client and 4.x client when authenticate fails,
When authenticate fails and the socket.io server throw an Error in middleware, e,g, `next(new Error("Invalid token"));` (refer to https://socket.io/docs/v4/middlewares/#sending-credentials). The 4.x client will stop reconnect. 

I think this is the connect behavior but for the 2.x client it will try to reconnect (at the next `pinginterval`) . But with the same incorrect credentials, the server will reject again but reconnect continues.

I think it is quite pointless to reconnect in this case so I think it is a bug for 2.x client code. But is there any particular reason why 2.x does it this way ?

**Comment by darrachequesne** [comment]
That sounds like a bug in the 2.x client. Let me check.

**Reply by qiulang** [reply]
When authenticate fails, just don't call `close()` on client side, then I see 2.x reconnects while 4.x stops. 

But if I call `close()` both 2.x & 4.x stop. So even it is a bug in 2.x it is not that a big deal.
